weave-js/src/components/PagePanelComponents/Home/Browse3/feedback/HumanFeedback/HumanFeedback.tsx

```diff
@@ -0,0 +1,680 @@
+import {toast} from '@wandb/weave/common/components/elements/Toast';
+import {MOON_300} from '@wandb/weave/common/css/color.styles';
+import {Button} from '@wandb/weave/components/Button';
+import {TextField} from '@wandb/weave/components/Form/TextField';
+import {LoadingDots} from '@wandb/weave/components/LoadingDots';
+import {Tailwind} from '@wandb/weave/components/Tailwind';
+import {parseRef} from '@wandb/weave/react';
+import _ from 'lodash';
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+
+import {CellValueString} from '../../../Browse2/CellValueString';
+import {useWFHooks} from '../../pages/wfReactInterface/context';
+import {useGetTraceServerClientContext} from '../../pages/wfReactInterface/traceServerClientContext';
+import {
+  FeedbackCreateError,
+  FeedbackCreateSuccess,
+} from '../../pages/wfReactInterface/traceServerClientTypes';
+import {
+  FEEDBACK_TYPE_OPTIONS,
+  HumanAnnotationPayload,
+  HumanFeedback,
+  makeAnnotationFeedbackType,
+  tsHumanAnnotationSpec,
+} from './humanFeedbackTypes';
+
+const DEBOUNCE_VAL = 200;
+
+// Interfaces
+type HumanFeedbackProps = {
+  entity: string;
+  project: string;
+  viewer: string | null;
+  hfSpec: tsHumanAnnotationSpec;
+  callRef: string;
+  readOnly?: boolean;
+  focused?: boolean;
+  setUnsavedFeedbackChanges: React.Dispatch<
+    React.SetStateAction<Record<string, () => Promise<boolean>>>
+  >;
+};
+
+export const HumanFeedbackCell: React.FC<HumanFeedbackProps> = props => {
+  const getTsClient = useGetTraceServerClientContext();
+  const tsClient = getTsClient();
+  const {useFeedback} = useWFHooks();
+  const [foundFeedback, setFoundFeedback] = useState<HumanFeedback[]>([]);
+  const query = useFeedback({
+    entity: props.entity,
+    project: props.project,
+    weaveRef: props.callRef,
+  });
+  const foundFeedbackCallRef = query?.result?.[0]?.weave_ref;
+  const feedbackSpecRef = props.hfSpec.ref;
+
+  useEffect(() => {
+    if (!props.readOnly) {
+      // We don't need to listen for feedback changes if the cell is editable
+      // it is being controlled by local state
+      return;
+    }
+    return getTsClient().registerOnFeedbackListener(
+      props.callRef,
+      query.refetch
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (foundFeedbackCallRef && props.callRef !== foundFeedbackCallRef) {
+      // The call was changed, we need to reset
+      setFoundFeedback([]);
+    }
+  }, [props.callRef, foundFeedbackCallRef]);
+
+  const onAddFeedback = async (value: number | string): Promise<boolean> => {
+    try {
+      const requestProps: FeedbackRequestProps = {
+        entity: props.entity,
+        project: props.project,
+        viewer: props.viewer,
+        callRef: props.callRef,
+        feedbackSpecRef,
+        value,
+      };
+
+      // TODO(gst): use replace when feedback is updated within 10 seconds of the previous feedback
```

Comment:
Shouldn't you always be using replace?

Reply:
With the "save" button I think we might not need replacing for v0. In the future allowing an explicit "edit" flow will use it.